node/src/BaseClient.ts

```diff
@@ -459,15 +508,21 @@ export class BaseClient {
             const errorType = getRequestErrorClass(message.requestError.type);
             reject(new errorType(message.requestError.message ?? undefined));
         } else if (message.respPointer != null) {
-            const pointer = message.respPointer;
+            let pointer;
 
-            if (typeof pointer === "number") {
-                // TODO: change according to https://github.com/valkey-io/valkey-glide/pull/2052
-                resolve(valueFromSplitPointer(0, pointer, true));
+            if (typeof message.respPointer === "number") {
```

Comment:
You have branching on pointer [type] here and on lines 627-638. Why not to fix up the pointer to have only one type and remove subsequent if-else blocks?

Reply:
Size matter. If we can use smaller size vars for each transaction (which can be a lot and hold most of the memory usage) I think it's better.